src/main/java/community/solace/spring/integration/leader/leader/SolaceLeaderInitiator.java

```diff
@@ -1,329 +1,305 @@
-import community.solace.spring.integration.leader.queue.ProvisioningException;
-import community.solace.spring.integration.leader.queue.SolaceLeaderViaQueue;
-import io.micrometer.core.instrument.Gauge;
-import io.micrometer.core.instrument.Metrics;
-import org.apache.commons.logging.Log;
-import org.apache.commons.logging.LogFactory;
-
-import org.springframework.boot.actuate.health.Health;
-import org.springframework.boot.actuate.health.HealthIndicator;
-import org.springframework.boot.context.event.ApplicationReadyEvent;
-import org.springframework.boot.context.event.ApplicationStartedEvent;
-import org.springframework.context.ApplicationContext;
-import org.springframework.context.ApplicationEventPublisher;
-import org.springframework.context.ApplicationEventPublisherAware;
-import org.springframework.context.event.EventListener;
-import org.springframework.integration.leader.Candidate;
-import org.springframework.integration.leader.Context;
-import org.springframework.integration.leader.DefaultCandidate;
-import org.springframework.integration.leader.event.DefaultLeaderEventPublisher;
-import org.springframework.integration.leader.event.LeaderEventPublisher;
-import org.springframework.jmx.export.annotation.ManagedAttribute;
-import org.springframework.jmx.export.annotation.ManagedOperation;
-import org.springframework.jmx.export.annotation.ManagedResource;
-import org.springframework.stereotype.Component;
-
-/**
- * Bootstrap leadership {@link org.springframework.integration.leader.Candidate candidates}
- * with Solace.
- * <p>
- * Mention that your queue failover timeout is configured at:
- * (configure/client-profile/service)# min-keepalive-timeout 10
- * on your broker.
- */
-@Component
-@ManagedResource()
-public class SolaceLeaderInitiator implements ApplicationEventPublisherAware, HealthIndicator {
-
-    private static final Log logger = LogFactory.getLog(SolaceLeaderInitiator.class);
-    private static final String SOLACE_GROUP_PREFIX = "leader.";
-    private final JCSMPSession session;
-    private final Map<String, LeaderGroupContainer> leaderGroups = new HashMap<>();
-    private final Map<String, LEADER_GROUP_JOIN> joinGroupsConfig;
-    private final Set<String> yieldOnShutdownConfig;
-    private final boolean anonymousGroupsArePermitted;
-    private final ApplicationContext appContext;
-
-    private Health.Builder health;
-    /**
-     * Leader event publisher.
-     */
-    private volatile LeaderEventPublisher leaderEventPublisher = new DefaultLeaderEventPublisher();
-
-    public SolaceLeaderInitiator(SpringJCSMPFactory solaceFactory, SolaceLeaderConfig solaceLeaderConfig, ApplicationContext appContext) {
-        this.joinGroupsConfig = SolaceLeaderConfig.getJoinGroupMap(solaceLeaderConfig);
-        this.yieldOnShutdownConfig = SolaceLeaderConfig.getYieldOnShutdown(solaceLeaderConfig);
-        this.anonymousGroupsArePermitted = solaceLeaderConfig.isPermitAnonymousGroups();
-        this.health = Health.down();
```

Comment:
Was that removed intentionally?

Reply:
The "this.health = Health.down()" -> yes, no need to start with health down since we get the connection already and if it's not connected the process won't start.